### PR TITLE
Create CSS-Animations.md

### LINKED ISSUE #681: CSS animations
I would like to add about CSS animations in `css`

### PR DESCRIPTION
**Pull request by AmrutaJayanti**
## Related Issue

Closes #681 

## Description

I added `CSS-Animations.md`

## Type of PR

- [ ] Bug fix
- [ ] Feature enhancement
- [x] Documentation update
- [ ] Other (specify): _______________

## Checklist
- [x] I have performed a self-review of my code.
- [x] I have read and followed the Contribution Guidelines.
- [x] I have tested the changes thoroughly before submitting this pull request.
- [x] I have provided relevant issue numbers, screenshots, and videos after making the changes.
- [x] I have commented my code, particularly in hard-to-understand areas.

## Resources for Guidance

Here are some resources that may be helpful as you contribute to CodeHarborHub:
- [Code Harbor Hub Community Features](https://www.codeharborhub.live/community/features)
- [Docusaurus Documentation](https://docusaurus.io/docs/create-doc)
- [React.js Documentation](https://legacy.reactjs.org/docs/getting-started.html)
- [Markdown Guide](https://www.markdownguide.org/)
- [MDX Documentation](https://mdxjs.com/docs/)
- [Mermaid Documentation](https://mermaid.js.org/)

**Resources for DSA Features on Our Site:**

- Tabs: [Switch Languages](https://www.codeharborhub.live/community/features#multi-language-support-code-blocks)
- [Line Numbering](https://www.codeharborhub.live/community/features#line-numbering)
- [Math Equations](https://www.codeharborhub.live/community/features#math-equations-with-katex)

### CODE REVIEW

File: docs/CSS/css-basics/CSS-Animations.md [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+/* The element to apply the animation to */
+div {
+  width: 100px;
+  height: 100px;
+  background-color: red;
+  animation-name: example;
+  animation-duration: 4s;
+}
+```
+<BrowserWindow url="http://127.0.0.1:5500/index.html">
```
Comment (Ajay-Dhangar) [inline docs/CSS/css-basics/CSS-Animations.md:37]: **Bad Practice**

```md
<BrowserWindow url="http://127.0.0.1:5500/index.html">
<div style="width: 100px; height: 100px; background-color: red; animation: my-animation 4s infinite; 
    @keyframes my-animation {
      from {background-color: red;}
      to {background-color: yellow;}
    }">
  Hello World!
</div>
</BrowserWindow>
```

**Good Practice**

```md
<BrowserWindow url="http://127.0.0.1:5500/index.html">
    <div style={{
      width: '100px',
      height: '100px',
      backgroundColor: 'red',
      animation: 'myAnimation 4s infinite',
      padding: '5px',
    }}>
      Hello World!
    </div>
    <style>{`
      @keyframes myAnimation {
        from { background-color: red; }
        to { background-color: yellow; }
      }
    `}</style>
  </BrowserWindow>
```

File: docs/CSS/css-basics/CSS-Animations.md [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+  width: 100px;
+  height: 100px;
+  position: relative;
+  background-color: red;
+  animation-name: example;
+  animation-duration: 4s;
+  animation-delay: 2s;
+}
+```
+<BrowserWindow url="http://127.0.0.1:5500/index.html">
```
Comment (Ajay-Dhangar) [inline docs/CSS/css-basics/CSS-Animations.md:65]: **Bad Practice**

```md
<BrowserWindow url="http://127.0.0.1:5500/index.html">
<div style="width: 100px;
  height: 100px;
  position: relative;
  background-color: red;
  animation-name: example;
  animation-duration: 4s;
  animation-delay: 2s;
  @keyframes example {
  0%   {background-color:red; left:0px; top:0px;}
  25%  {background-color:yellow; left:200px; top:0px;}
  50%  {background-color:blue; left:200px; top:200px;}
  75%  {background-color:green; left:0px; top:200px;}
  100% {background-color:red; left:0px; top:0px;}
}"
>
  Hello World!
</div>
</BrowserWindow>
```

**Good Practice**

```md
<BrowserWindow minHeight={400} url="http://127.0.0.1:5500/index.html">
    <div className="Delay-an-Animation" style={{
      width: '100px',
      height: '100px',
      position: 'relative',
      backgroundColor: 'red',
      animationName: 'DelayAnAnimationExample',
      animationDuration: '4s',
      animationDelay: '2s',
      padding: '5px',
    }}>
      Hello World!
    </div>
    <style>{`
      @keyframes DelayAnAnimationExample {
        0%   { background-color: red; left: 0px; top: 0px; }
        25%  { background-color: yellow; left: 200px; top: 0px; }
        50%  { background-color: blue; left: 200px; top: 200px; }
        75%  { background-color: green; left: 0px; top: 200px; }
        100% { background-color: red; left: 0px; top: 0px; }
      }
      .Delay-an-Animation {
        position: absolute;
      }
    `}</style>
  </BrowserWindow>
```

File: docs/CSS/css-basics/CSS-Animations.md [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+  width: 100px;
+  height: 100px;
+  position: relative;
+  background-color: red;
+  animation-name: example;
+  animation-duration: 4s;
+  animation-iteration-count: 3;
+}
+```
+
```
Comment (Ajay-Dhangar) [inline docs/CSS/css-basics/CSS-Animations.md:99]: ### Bad Practice

```md
<BrowserWindow url="http://127.0.0.1:5500/index.html">
<div style="width: 100px;
  height: 100px;
  position: relative;
  background-color: red;
  animation-name: example;
  animation-duration: 4s;
  animation-iteration-count: 3;
  @keyframes example {
  0%   {background-color:red; left:0px; top:0px;}
  25%  {background-color:yellow; left:200px; top:0px;}
  50%  {background-color:blue; left:200px; top:200px;}
  75%  {background-color:green; left:0px; top:200px;}
  100% {background-color:red; left:0px; top:0px;}
}"
>
  Hello World!
</div>
</BrowserWindow>
```

### Good Practice

```md
<BrowserWindow minHeight={400} url="http://127.0.0.1:5500/index.html">
    <div style={{
      width: '100px',
      height: '100px',
      position: 'relative',
      backgroundColor: 'red',
      animationName: 'example',
      animationDuration: '4s',
      animationIterationCount: 3,
      padding: '5px',
    }}>
      Hello World!
    </div>
    <style>{`
      @keyframes example {
        0% { background-color: red; left: 0px; top: 0px; }
        25% { background-color: yellow; left: 200px; top: 0px; }
        50% { background-color: blue; left: 200px; top: 200px; }
        75% { background-color: green; left: 0px; top: 200px; }
        100% { background-color: red; left: 0px; top: 0px; }
      }
    `}</style>
  </BrowserWindow>
```

File: docs/CSS/css-basics/CSS-Animations.md [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+  height: 100px;
+  position: relative;
+  background-color: red;
+  animation-name: example;
+  animation-duration: 4s;
+  animation-direction: reverse;
+}
+```
+
+<BrowserWindow url="http://127.0.0.1:5500/index.html">
```
Comment (Ajay-Dhangar) [inline docs/CSS/css-basics/CSS-Animations.md:143]: ### Bad Practice

```md
<BrowserWindow url="http://127.0.0.1:5500/index.html">
<div style=" width: 100px;
  height: 100px;
  position: relative;
  background-color: red;
  animation-name: example;
  animation-duration: 4s;
  animation-direction: reverse;
  @keyframes example {
  0%   {background-color:red; left:0px; top:0px;}
  25%  {background-color:yellow; left:200px; top:0px;}
  50%  {background-color:blue; left:200px; top:200px;}
  75%  {background-color:green; left:0px; top:200px;}
  100% {background-color:red; left:0px; top:0px;}
}"
>
  Hello World!
</div>
</BrowserWindow>
```

### Good Practice

```mdx
<BrowserWindow minHeight={400} url="http://127.0.0.1:5500/index.html">
    <div style={{
      width: '100px',
      height: '100px',
      position: 'relative',
      backgroundColor: 'red',
      animationName: 'example',
      animationDuration: '4s',
      animationDirection: 'reverse',
      padding: '5px',
    }}>
      Hello World!
    </div>
    <style>{`
      @keyframes example {
        0%   { background-color: red; left: 0px; top: 0px; }
        25%  { background-color: yellow; left: 200px; top: 0px; }
        50%  { background-color: blue; left: 200px; top: 200px; }
        75%  { background-color: green; left: 0px; top: 200px; }
        100% { background-color: red; left: 0px; top: 0px; }
      }
    `}</style>
  </BrowserWindow>
```

**Review by Ajay-Dhangar — CHANGES_REQUESTED** [review]
Now update code based on suggestions

**Comment by AmrutaJayanti** [comment]
@Ajay-Dhangar I have done the changes you suggested. Please approve

**Review by Ajay-Dhangar — APPROVED** [review]
Done

**Pull request merged**